gytheio-content-handlers/gytheio-content-handler-s3/src/main/java/org/gytheio/content/handler/s3/S3ContentReferenceHandlerImpl.java

```diff
@@ -281,15 +295,75 @@ public InputStream getInputStream(ContentReference contentReference, boolean wai
         }
     }
 
+    @Override
+    public long putFile(File sourceFile, ContentReference targetContentReference)
+            throws ContentIOException
+    {
+        if (! isContentReferenceSupported(targetContentReference))
+        {
+            throw new ContentIOException("ContentReference not supported");
+        }
+
+        long startTime = System.currentTimeMillis();
+
+        String remotePath = getRelativePath(targetContentReference.getUri());
+
+        try
+        {
+            Long contentLength = sourceFile.length();
+
+            try
+            {
+                Upload xfer = tm.upload(s3BucketName, remotePath, sourceFile);
+                xfer.waitForCompletion();
+            }
+            catch (InterruptedException e)
+            {
+                logger.error("Upload was interrupted: "+e.getMessage());
+
+                // Be a good citizen  and set interrupt flag
+                Thread.currentThread().interrupt();
+
+                throw new ContentIOException("Failed to write content", e);
+            }
+
+            ObjectMetadata metadata = s3.getObjectMetadata(
+                    new GetObjectMetadataRequest(s3BucketName, remotePath));
+
+            long storedContentLength = metadata.getContentLength();
+
+            if (logger.isWarnEnabled())
+            {
+                if ((contentLength != null) && (storedContentLength != contentLength))
```

Comment:
**contentLength** is always different than **null**. In fact it can be declared as a '**long**' value (primitive), as it's the type returned by _sourceFile.length()_.